src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1456,6 +1518,15 @@ MulticopterPositionControl::task_main()
 			_alt_hold_engaged = false;
 		}
 
+		// assume really landed if landed, and if we have seen landed state for at least 0.5 seconds
+		bool landed_for_sure = _vehicle_land_detected.landed && _time_landed != 0 && (hrt_elapsed_time(&_time_landed) > 5e5);
+
+		// once we are sure that we are landed, then set the landed lock which prevents noise from the landed detector
+		// to interrupt the takeoff
+		if (landed_for_sure && !_landed_locked) {
```

Comment:
@tumbili do we really need yet another delay here? the land detection has that already.